src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -444,38 +442,101 @@ def _set_energy_forecast(self, payload):
                 IncomingRequest("set_energy_forecast", arguments,
                                 energy_forecast_response_channel))
 
-    def _set_energy_forecast_impl(self, arguments, response_channel):
+    def _set_energy_measurement(self, payload: Dict) -> None:
```

Comment:
Nit: if we want to follow SOLID principles, we should probably think about refactoring the `ExternalMixin` class (and the other strategies). The `_set_energy_forecast/measurement` methods, for example, are defined in the `ExternalMixin` and inherited in all the external strategies, even though they're not used (for example in the `StorageExternalMixin`).

Reply:
Since this method is called from other classes, I think it should be a public method (without the leading underscore). The same applies to other methods in the same class, in case you are doing some refactoring.

Reply:
> Nit: if we want to follow SOLID principles, we should probably think about refactoring the `ExternalMixin` class (and the other strategies). The `_set_energy_forecast/measurement` methods, for example, are defined in the `ExternalMixin` and inherited in all the external strategies, even though they're not used (for example in the `StorageExternalMixin`).

Yeah, Hassan proposed to create yet another base class, but I like your proposal. Will add these to the `ExternalMixin`. thanks

Reply:
Wait, what would you add to the `ExternalMixin`? My point was exactly that the `ExternalMixin` contains too many methods and that these methods are not being used in all the classes that inherit the `ExternalMixin` (and that's not good) :D

Reply:
Upsi, sorry, read too fast. Here we start a gain the never ending discussion about whether and how to handle the ExternalMixin. What would be your proposal of refactoring the ExternalMixin? Having one mixing for each device type? 
In any case, lets create a separate story for that. This settlement market is dragging too long already, we need to get it done soon. OK?

Reply:
Definitely: working on the `ExternalMixin` deserves a separate story (and probably some conversation to agree on a new structure). I'll create a task on Jira.